cmake/python.cmake

```diff
@@ -1,7 +1,13 @@
 # the CMake variable PYTHON_INSTALL_DIR has the same value as the Python function catkin.builder.get_python_install_dir()
 
 set(PYTHON_VERSION "$ENV{ROS_PYTHON_VERSION}" CACHE STRING "Specify specific Python version to use ('major.minor' or 'major')")
-find_package(PythonInterp ${PYTHON_VERSION} REQUIRED)
+find_package(Python ${PYTHON_VERSION} REQUIRED)
+
+# Set these legacy variables for compatibility with what PythonInterp used to do
+set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
```

Comment:
I don't think it's safe to just migrate to the new variable names— the ROS 1 codebase is pretty dependent on `PYTHON_EXECUTABLE` in particular existing, see for example the stuff around message generation:

- https://github.com/ros/genmsg/blob/noetic-devel/cmake/pkg-genmsg.cmake.em
- https://github.com/ros/dynamic_reconfigure/blob/noetic-devel/cmake/dynamic_reconfigure-macros.cmake

Amusingly, dynamic_reconfigure does actually call `find_package(PythonInterp)`, but does so _after_ `PYTHON_EXECUTABLE` has already been used once already.